R/dist.R

```diff
@@ -1382,48 +1474,374 @@ plot.dist_spec <- function(x, ...) {
+##'
+##' @inheritParams stats::Lognormal
+##' @param mean,sd mean and standard deviation of the distribution
+##' @param max Numeric, maximum value of the distribution. The distribution will
+##' be truncated at this value. Default: `Inf`, i.e. no maximum.
+##' @return A `dist_spec` representing a distribution of the given
+##'   specification.
+##' @author Sebastian Funk
+##' @export
+##' @rdname Distributions
+##' @name Distributions
+##' @order 1
+##' @examples
+##' LogNormal(mean = 4, sd = 1)
+##' LogNormal(mean = 4, sd = 1, max = 10)
+##' LogNormal(mean = Normal(4, 1), sd = 1, max = 10)
+LogNormal <- function(meanlog, sdlog, mean, sd, max = Inf) {
+  params <- as.list(environment())
+  return(new_dist_spec(params, "lognormal"))
+}
+
+##' @inheritParams stats::GammaDist
+##' @author Sebastian Funk
+##' @rdname Distributions
+##' @title Probability distributions
+##' @order 2
+##' @export
+##' @examples
+##' Gamma(mean = 4, sd = 1)
+##' Gamma(shape = 16, rate = 4)
+##' Gamma(shape = Normal(16, 2), rate = Normal(4, 1))
+Gamma <- function(shape, rate, scale, mean, sd, max = Inf) {
+  params <- as.list(environment())
+  return(new_dist_spec(params, "gamma"))
+}
+
+##' @rdname Distributions
+##' @order 3
+##' @author Sebastian Funk
+##' @export
+##' @examples
+##' Normal(mean = 4, sd = 1)
+##' Normal(mean = 4, sd = 1, max = 10)
+Normal <- function(mean, sd, max = Inf) {
+  params <- as.list(environment())
+  return(new_dist_spec(params, "normal"))
+}
+
+##' @rdname Distributions
+##' @order 4
+##' @param value Value of the fixed (delta) distribution
+##' @author Sebastian Funk
+##' @export
+##' @examples
+##' Fixed(value = 3)
+##' Fixed(value = 3.5)
+Fixed <- function(value, max = Inf) {
+  params <- as.list(environment())
+  return(new_dist_spec(params, "fixed"))
+}
+
+##' Generates a nonparametric distribution.
+##'
+##' @param mass Probability mass of the given distribution; this is
+##'   passed as a zero-indexed numeric vector (i.e. the fist entry represents
+##'   the probability mass of zero). If not summing to one it will be normalised
+##'   to sum to one internally.
+##' @author Sebastian Funk
+##' @rdname Distributions
+##' @order 5
+##' @export
+##' @examples
+##' pmf(c(0.1, 0.3, 0.2, 0.4))
+##' pmf(c(0.1, 0.3, 0.2, 0.1, 0.1))
+pmf <- function(mass) {
+  params <- list(pmf = mass / sum(mass))
+  return(new_dist_spec(params, "nonparametric"))
+}
+
+##' Get the names of the natural parameters of a distribution
+##'
+##' These are the parameters used in the stan models. All other parameter
+##' representations are converted to these using [convert_to_natural()] before
+##' being passed to the stan models.
+##' @param distribution Character; the distribution to use.
+##' @return A character vector, the natural parameters.
+##' @author Sebastian Funk
+##' @keywords internal
+##' @examples
+##' \dontrun{
+##' natural_params("gamma")
+##' }
+natural_params <- function(distribution) {
```

Comment:
dist is very long and hard to pass. I think it might be better to split into dist-tools and dist

Reply:
Split now into:
- `dist_spec.R`: everything related to `<dist_spec>`. This should now not use any other code from the package so could in principle be extracted.
- `deprecated.R`: deprecated functions (could add other ones from the rest of the package here later to improve code readability)
- `estimate_delay.R`: everything related to estimating delays
- moved `sample_approx_dist` to `adjust_infection_to_report` as it's the only place where it's used. Eventually they might be deprecated together as suggested in #520